wolfSSL/src/chapter06.md

```diff
@@ -185,3 +185,158 @@ These contexts may be pointers to any user-specified context, which will then in
 Example callbacks can be found in `wolfssl/test.h`, under `myEccSign()`, `myEccVerify()`, `myEccSharedSecret()`, `myRsaSign()`, `myRsaVerify()`, `myRsaEnc()`, and `myRsaDec()`.  Usage can be seen in the wolfSSL example client (`examples/client/client.c`), when using the `-P` command line option.
 
 To use Atomic Record Layer callbacks, wolfSSL needs to be compiled using the [`--enable-pkcallbacks`](chapter02.md#--enable-pkcallbacks) configure option, or by defining the `HAVE_PK_CALLBACKS` preprocessor flag.
+
+
+## Crypto Callbacks (cryptocb)
+
+The Crypto callback framework in wolfSSL/wolfCrypt enables users to override the default implementation of select cryptographic algorithms and provide their own custom implementations at runtime. The most common use case for crypto callbacks is to offload an algorithm to custom hardware acceleration,  however it could also be used to add additional logging/introspection, retrieve keys from secure storage, to call crypto from another library for standards compliance reasons, or even to perform a remote procedure call. 
+
+### Using Crypto callbacks
+
+The high level steps use crypto callbacks are:
+
+1. Compile wolfSSL with crypto callback support
+2. Register a callback and unique device ID (`devId`) with wolfCrypt
+3. Pass the `devId` as an argument to a wolfCrypt API function
+4. (TLS only): associate the `devId` with a wolfSSL context
+
+
+#### 1. Compile wolfSSL with crypto callback support
+Support for crypto callbacks is enabled in wolfSSL via the `–enable-cryptocb` configure option, or `#define WOLF_CRYPTO_CB`.
+
+#### 2. Register your callback and unique devID with wolfCrypt
+
+To use crypto callbacks, the user must register a top-level cryptographic callback function with wolfCrypt and provide a device ID (`devId`), which is an integer ID that uniquely identifies this particular callback function to wolfCrypt. This allows multiple crypto callbacks to be registered with wolfCrypt at once. The appropriate callback function will then be invoked internally by every wolfCrypt function that takes a `devId` parameter, or by wolfSSL when associated with a `WOLFSSL_CTX` or `WOLFSSL` structure.
```

Comment:
Please add comment about `INVALID_DEVID` and it being `-2`.

Reply:
done